costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -127,25 +126,11 @@ Costmap2DROS::Costmap2DROS(std::string name, tf::TransformListener& tf) :
   }
 
   // subscribe to the footprint topic
-  std::string topic_param, topic;
-  if(!private_nh.searchParam("footprint_topic", topic_param))
-  {
-    topic_param = "footprint_topic";
-  }
-
-  private_nh.param(topic_param, topic, std::string("footprint"));
-  footprint_sub_ = private_nh.subscribe(topic, 1, &Costmap2DROS::setUnpaddedRobotFootprintPolygon, this);
+  footprint_sub_ = private_nh.subscribe("footprint_spec", 1, &Costmap2DROS::setUnpaddedRobotFootprintPolygon, this);
```

Comment:
the parameter here might be valuable -- what if you wanted to use different footprint topics for different costmaps within the same node?